app/scripts/metamask-controller.test.js

```diff
@@ -642,6 +642,40 @@ describe('MetaMaskController', () => {
       });
     });
 
+    describe('#createNewVaultAndRestore - Nonce Detection', () => {
+      it('should continue with accounts having non-zero nonce', async () => {
+        const password = 'password123';
+        jest.spyOn(metamaskController, 'getBalance').mockResolvedValue('0x0');
+        jest.spyOn(metamaskController, 'getNonce').mockResolvedValue('0x1'); // Mock a non-zero nonce
+
+        // Call the restore function
+        await metamaskController.createNewVaultAndRestore(password, TEST_SEED);
+
+        // Expectation: The restoration should not stop at the first account with zero balance
+        // You need to adjust this based on your implementation details
+        expect(
+          metamaskController.keyringController.addNewAccount,
+        ).toHaveBeenCalled();
+      });
+    });
+
+    describe('#createNewVaultAndRestore - Account Skipping', () => {
```

Comment:
Optional: we could merge the tests under `#createNewVaultAndRestore` and leave 2 scenarios under. wdyt?